habitat_sim/simulator.py

```diff
@@ -254,19 +239,10 @@ def __set_from_config(self, config: Configuration) -> None:
 
         self._default_agent_id = config.sim_cfg.default_agent_id
 
-        self.__sensors: List[Dict[str, Sensor]] = [
-            dict() for i in range(len(config.agents))
-        ]
         self.__last_state = dict()
         for agent_id, agent_cfg in enumerate(config.agents):
             for spec in agent_cfg.sensor_specifications:
-                self._update_simulator_sensors(spec.uuid, agent_id=agent_id)
-            self.initialize_agent(agent_id)
-
-    def _update_simulator_sensors(self, uuid: str, agent_id: int) -> None:
-        self.__sensors[agent_id][uuid] = Sensor(
-            sim=self, agent=self.get_agent(agent_id), sensor_id=uuid
-        )
+                self.add_sensor(spec, agent_id=agent_id)
```

Comment:
The argument order is really weird. 
The natural way should be agent id first, then the spec.